react/Notices.tsx

```diff
@@ -20,6 +20,20 @@ const Notices: FC = () => {
     return null
   }
 
+  orders.forEach((order) => {
+    order.paymentData.transactions.sort((transaction) =>
+      transaction.payments.some((payment) => payment.group === 'bankInvoice')
+        ? -1
+        : 1
+    )
+
+    order.paymentData.transactions.forEach((transaction) => {
+      transaction.payments.sort((payment) =>
+        payment.group === 'bankInvoice' ? -1 : 1
+      )
+    })
+  })
+
```

Comment:
```suggestion
 const sortTransactions = (transactions) => {
  return transactions.slice().sort((a, b) => {
    const hasBankInvoiceA = a.payments.some((p) => p.group === 'bankInvoice');
    const hasBankInvoiceB = b.payments.some((p) => p.group === 'bankInvoice');
    return hasBankInvoiceA === hasBankInvoiceB ? 0 : hasBankInvoiceA ? -1 : 1;
  });
};

const sortPayments = (payments) => {
  return payments.slice().sort((a, b) => (a.group === 'bankInvoice') - (b.group === 'bankInvoice'));
};

const updatedOrders = orders.map((order) => ({
...order,
  paymentData: {
...order.paymentData,
    transactions: sortTransactions(order.paymentData.transactions).map((transaction) => ({
...transaction,
      payments: sortPayments(transaction.payments),
    })),
  },
}));

```

Mantendo imutabilidade da variável original

Reply:
Obrigado pelas sugestões.
Em meu novo commit eu fiz apenas alguns ajustes em cima da sua sugestão:

- Removi os nested ternaries da linha 69;
- Corrigi a função sortPayments(), que estava invertida e o TypeScript não aceitou a operação aritmética entre booleanos.